src/mango/src/mango_selector.erl

```diff
@@ -1070,12 +1073,18 @@ check_beginswith(Field, Prefix) ->
 match_beginswith_test() ->
     % matching
     ?assertEqual(true, check_beginswith(<<"_id">>, <<"f">>)),
-    % no match (user_id is not a binary string)
+    % no match (user_id field in the test doc contains an integer)
     ?assertEqual(false, check_beginswith(<<"user_id">>, <<"f">>)),
     % invalid (prefix is not a binary string)
     ?assertThrow(
         {mango_error, mango_selector, {invalid_operator, <<"$beginsWith">>}},
         check_beginswith(<<"user_id">>, 1)
+    ),
+    % invalid (prefix is not a utf8 string)
+    InvalidArg = <<32#FFFFF>>,
```

Comment:
A tiny nit: this would just be `<<16#EF>>` as is we're creating a base 32 larger integer, but only the last byte is used from it.

```erlang
 <<16#EF>> =:= <<32#FFFFF>>.
true
```